pkgs/tools/networking/curl-impersonate/deps.nix

```diff
@@ -0,0 +1,29 @@
+# Generated by update.py
+{ fetchurl }:
+
+{
+  "curl-7.84.0.tar.xz" = fetchurl {
+    url = "https://curl.se/download/curl-7.84.0.tar.xz";
+    hash = "sha256-LRGLQ/VHv+W66AbY1HtOWW6lslpsHwgK70n7zYF8Xbg=";
+  };
+
+  "brotli-1.0.9.tar.gz" = fetchurl {
+    url = "https://github.com/google/brotli/archive/refs/tags/v1.0.9.tar.gz";
+    hash = "sha256-+ejYHQQFumbRgVKa9CozVPg4yTkJX/mZMNpqqc32/kY=";
+  };
+
+  "nss-3.77.tar.gz" = fetchurl {
+    url = "https://ftp.mozilla.org/pub/security/nss/releases/NSS_3_77_RTM/src/nss-3.77-with-nspr-4.32.tar.gz";
+    hash = "sha256-j6UEIeWcTk5x0Z8j3WV3uK+xwT4vm38OyxpVrjd1KV8=";
+  };
+
+  "boringssl.zip" = fetchurl {
+    url = "https://github.com/google/boringssl/archive/3a667d10e94186fd503966f5638e134fe9fb4080.zip";
+    hash = "sha256-HsDIkd1x5IH49fUF07dJaabMIMsQygW+NI7GneULpA8=";
+  };
+
+  "nghttp2-1.46.0.tar.bz2" = fetchurl {
+    url = "https://github.com/nghttp2/nghttp2/releases/download/v1.46.0/nghttp2-1.46.0.tar.bz2";
+    hash = "sha256-moKXjIcAcbdp8n0riBkct3/clFpRwdaFx/YafhP8Ryk=";
+  };
+}
```

Comment:
Why do we vendor those downloads instead of using ours?

Reply:
I would worry if we changed the versions too much, the TLS fingerprints may no longer match the browsers it is supposed to be impersonating or the patches from the project may not apply cleanly. Admittedly I did not test that theory, so it is possible using our dependencies could be fine

Reply:
Hi, I just noticed this thread (I'm the author of curl-impersonate). I believe you can use whatever nghttp2 and brotli versions you'd like without it affecting the TLS signatures. I didn't see a case yet where it mattered (when I started curl-impersonate I didn't know that hence the fixed versions). Regarding nss and boringssl, it is important to use known working versions, otherwise the TLS signature might change (new TLS extensions, different default ciphers, and so on).